runtime/src/main/scala/akka/grpc/scaladsl/headers/PercentEncoding.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2018-2021 Lightbend Inc. <http://www.lightbend.com>
+ * Copyright 2016, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package akka.grpc.scaladsl.headers
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+
+/**
+ * Excerpt from https://github.com/grpc/grpc/blob/7c9e8b425166276232653725de32ea0422a39b33/doc/PROTOCOL-HTTP2.md#responses:
+ *
+ * The value portion of Status-Message is conceptually a Unicode string description of the error, physically encoded as
+ * UTF-8 followed by percent-encoding. Percent-encoding is specified in RFC 3986 §2.1, although the form used here has
+ * different restricted characters. When decoding invalid values, implementations MUST NOT error or throw away the
+ * message. At worst, the implementation can abort decoding the status message altogether such that the user would
+ * received the raw percent-encoded form. Alternatively, the implementation can decode valid portions while leaving
+ * broken %-encodings as-is or replacing them with a replacement character (e.g., '?' or the Unicode replacement
+ * character).
+ */
+private[grpc] object PercentEncoding {
+  // Copied with slight adaptations from https://github.com/grpc/grpc-java/blob/79e75bace40cea7e4be72e7dcd1f41c3ad6ee857/api/src/main/java/io/grpc/Status.java#L582
+  object Encoder {
+    private val HexArr: Array[Byte] =
+      Array('0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F')
+
+    def encode(value: String): String = {
+      val valueBytes = value.getBytes(StandardCharsets.UTF_8)
+      val firstIndexToEscape = valueBytes.indexWhere(isEscapingChar(_))
+      if (firstIndexToEscape == -1)
+        new String(valueBytes)
```

Comment:
Missing character encoding in String constructor here

Reply:
Corrected